doc/ref_impl/cntt-ri/chapters/chapter04.md

```diff
@@ -7,56 +7,62 @@
 
+The lab provides a set of bare-metal servers, supported by a dedicated network infrastructure.  Users should be provided access to the servers, their operating systems, and out-of-band management (i.e. IPMI, IDRAC, ILO, etc.).  
 
 <a name="4.2"></a>
 ## 4.2 Lab Requirement
 
-- Controller Nodes:
-  - 3 x
-    - 2x dual-port 10Gbps NIC.
-    - 2.2GHz 14C/28T.
-    - 256GB RAM.
-    - 10TB HDD.
-    - 3.2TB SSD.
-- Compute Node
-  - 10x
-    - 2x dual-port 25Gbps NIC.
-    - 2.2GHz 24C/48T.
-    - 512GB RAM.
-    - 2TB HDD
-    - Supports 2x Nodes for B/N/C flavors
-- Jump/Baremetal manager
-  - 1x
-    - 2x dual-port 10Gbps NIC.
-    - 2.2GHz 14C/28T.
-    - 256GB RAM.
-    - 10TB HDD
-- Networking
-  - 1x Spine Switch
-    - Total: 32x100G
-  - 2x Leaf Switch
-    - 48 x 25/10G
-    - 6 x 100G
-- 1x 48u Rack
-- Cables and Transcievers
+Labs are organized into one or more *Pods*, where the *Pod* provides the complete set of resources [servers and network(s)] necessary to support the reference implementation installation and operation.  
+
+A CNTT compliant *Pod* shall meet the following requirements.
+
+1. One (1) bare-metal server dedicated as a Jump / Test Host
+2. Six (6) bare-metal servers, serving as either compute or controllers
+3. A configured network topology allowing for: Out-of-Band Management, Admin, Public, Private, and Storage Networks
+
+The Jump Host / Test Host system will be utilized to install and manage the operation of the *Pod*.  For example, the system can be used to host the installer software used to deploy and configure OpenStack on the other bare-metal servers.  Once installed, the system may be used to run test automations or it may be used to generate test traffic intended to measure the performance of a VNF operating within the *Pod*.  To support these operations, the bare-metal server provided for the Jump / Test Host shall meet the same hardware requirements as compute / controller, as outline below.  In this context, the system may be considered as a miscellaneous use  or general purpose system.
+
+<a name="4.2.1"></a>
+### 4.2.1 Bare-metal Server Requirements
+Each server shall meet the following minimum specifications:
+
+- **CPU**
+  - 2x x86_64 CPU sockets, providing 24 cores each, 44 hyper-threads, at 2.2 GHz
+- **Memory**
+  - 512 GB RAM
+- **Storage**
+  - 3.2 TB SSD via SATA 6 Gbps
+  - Storage should present as at least 3 or more disks to the OS, allowing for usage as CEPH storage nodes, or similar.
+- **Network Interfaces** (note 1)
+  - 4x 10Gbps Ethernet Ports
+  - Out-of-band Management Port
+
+Note 1: At least 1 network interface must be capable of performing PXE boot and that network must be available to both the Jump / Test Host and each bare-metal server.
+
+<a name="4.2.2"></a>
+### 4.2.2 Networking Hardware
```

Comment:
we specified Spine/Leaf switches in the org specs as this is the model assumed in RA 1. I know there are contributions in the pipelines to do that. can we add more information here and can we make sure we adopt Spine/Leaf architectures and have spec the networking devices?

Reply:
@rabi-abdel By networking devices, you're referring to the leaf and spine switches? Are you envisioning make and model, or physical description (i.e. ports/speeds). In the interest of neutrality, I'd recommend the latter.

@ASawwaf Included in the leaf/spine spec, we need to align on the/any required APIs we require. For example, must provide open/public APIs for <list>, where $list might include NC, P4, OF or whatever. 

Again, networking should not gate server approval, except for NICs, which has already been resolved (above), and vice versa.

Reply:
It's likely not practical to require the switches support a superset of SDN protocols.  It would be hard to fine a switch supporting both P4 and OpenFlow for example.  I'm not sure how we provide flexibility to the RA/RM definitions here, while documenting the requirements for lab hardware and implementation.  Unless the Reference Implementation will make a specific choice, and the lab is then required to support that choice.

Reply:
@markshostak it is the latter

Reply:
Agreed, and what I listed was to illustrate what I meant by APIs. I'm guessing the fabric will use distributed control, so really only NC would be relevant, if that, but the point is we need to align on what we require. I don't have a specific preference at this time, but Ahmed will.

Reply:
On the leaf / spine topology.  I'm struggling with how to fit that into the context of the Pod definition.  For example, the Pod, as defined only has 7 total servers.  To provide an independent leaf / spine per pod would really require 2 leaf and 2 spine switches, which doesn't scale.  We could provide guidance for labs like LaaS to ensure the network(s) supporting multiple Pods meets the leaf / spine topology requirement.

Reply:
@ASawwaf agree with you on the need for this, but for the purpose of this PR, I think it is ok to only specify high level specs with the acknowledgment that we will be specifying them in more details as the specs matures and evolve.

Reply:
@rabi-abdel Are you referring to the API thing, when you say only specify high level specs?

Reply:
@markshostak Yes API but pithing drilling to too much into details. (i.e supporting BGP protocols, encapsulation, etc). @ASawwaf probably can help specifying that.

Reply:
@lylavoie I don't think the intent here is to provide an independent leaf/spine to each pod. the pod is a special use case in my views. but the lab as a whole, we require a spine/lead architecture. I hope that makes sense.

Reply:
@rabi-abdel that makes complete sense, and I can add text to make that clear.  I just don't want to set an expectation that each pod (which is kinda of a "unit of access" or a lab deliverable to the users) gets an independent set of lead/spine switches, as that would be unrealistic.

Reply:
@lylavoie Lincoln, I agree with Rabi. A reasonable question for us to run down for you is, is a single spine sufficient or are 2 required. However, the driver for that decision is irrespective of the number of PODs. The intent is to have a shared fabric and definitely **not** to have a dedicated physical leaf/spine for _each_ POD. Whether to have 2 spines and 2 leafs in total, or 1 and 1 (provided the leaf has sufficient ports for all of the servers), is where I see the questions. 

To that point, as is probably obvious, the driver for the NIC config we've discussed (2x dual 25Gb (4 ports)), is to have 1 port from each NIC go to one leaf and the other port from the same NIC go to the other leaf. That would imply we should have a min of 2 physical leaf switches (more, added in pairs, if required to provide the number of ports needed for the number of servers in the infra).

Whether to connect the leaf switches with 1 or 2 spines, is a different question. I'll defer to Ahmed @ASawwaf , for direction there.

Irrespective, as Rabi implied, all of the servers would share the fabric described above. Coexistence will be provided by isolation and encapsulation protocols. We can pursue this further in a dedicated fabric discussion, which is out of scope here, but we'd like to hear how LaaS and/or your IOL currently provides isolation, if there's any encapsulation and if you're fabric is L2 or L3 based. From there we can determine the actual fabric topology and design (and get Rabi his diagram), and what, if any enhanced support will be required from the LaaS automation. I can envision ways we can leverage additional LaaS features, and ways we could get by without.

Does that answer your question/clarify the vision?

So, we owe you:
1) number of spines
2) if leaf switches need to be paired
@ASawwaf Can you give us your input on the 2 above? Thanks

Reply:
@markshostak with respect to LaaS, we current provide isolation at Layer 2 using VLANs. 

For the question about "paired" leaf switches, I could go either way.  It might make sense, if we are going to start testing HA or failing a link, etc.  It's not really practical to test failing a switch in this environment, if the switch is supporting multiple pods and you're "neighbors" aren't expecting to test that case at that minute.  

Same for number of spine switches.  I think could be fine to role out with one for release 1, and add more as a requirement, if we see the need.

Reply:
@lylavoie @markshostak @rabi-abdel 

The answers for the above 2 points will be on 
1-  # of servers  
2- Use case

for lab purpose, if we have one RACK only, we can leave with Single SW covers both Spine/Leaf
but this SW should have enough capacity 

To have the right decision 
@liyingyjy can share what we have or what can we secure in terms of 
- number of servers 
- Which server vendor that we have / or will have
- which NIC card vendor 
- Which SW vendors

@liyingyjy for VLAN separation, it is not a recommendation to have VLAN inside the Fabric 



My comments and recommendation 
-
- We need to lock down the use case or SUT that should covers ( i.e Just VNF onboarding to match the profile or VNF performance, or ...
- To have a Small mimic of what the operator should implement 

- # of server : 3 Server ( OSP + SDN-C ) + 1 Server ( Jump+Installer ) + 3 CEPH + Computer/workload : at least to 2 per profile ( 6 Server ) 
- Sequential VNF testing may be a challenge due to the limitation of workload servers 
- NIC Card either be 10G / 25G based on the use case ( for scalability 25G ) , Just want to unify the Fabric ( either 10G or 25G )
- The server should be NIC-NUMA aligned for VNF at least 
- We can host the Openstack controller with SDN -C ( as @markshostak mentioned before ) 
- RAM : Should be balanced between CPU ( no matter if it is 32G / 16G ) but should be balanced
- CEPH : we need to have min 3 CEPH Cluster, my recommendation to go with SSD for High IOPS req.
- We need to cover Jump server, Installer Server and Tools servers ( Servers or VM )

Reply:
I don't know why bold font appears  :):)

Reply:
My 2 pence - let's focus on the immediate objective - **align on the server hardware config, so we can approve it on Wednesday's TSC call** and once that's locked, for extra credit we may provide some guidelines on the fabric. 

I agree with all of the statements above, but there's no way we can resolve them in time for the TSC call. The latest server configuration proposed seems like it meets our gross requirements. No one I'm aware of has identified an [open] issue with the current config. @kedmison identified a risk WRT clock speed, in that the minimum speed (or if clock speed will be specified) hasn't been determined yet, hence if these servers end up not meeting the final clock spec, that could be an issue. I'd like to know the impact if this risk is realized. My working assumption is even if they're not quite as fast as the spec, they'll still be faster than if we miss the window and don't get servers. @lylavoie also needs to weigh in with if this is something UNH is comfortable with. Kelvin, Lincoln, thoughts? Thx, -M

Reply:
I've added text for the switches (i.e. leaf / spine).

For the CPU topics, I'm happy with the current definition, as the minimum specs for release 1.  It's challenging to use other specifications, while still remaining general in the requirements (i.e. non product specific).